### PR TITLE
widgets: time-series: responsive

### PR DESCRIPTION
**Pull request by matallo**
re: https://github.com/CartoDB/cartodb/issues/12470

this will eventually point to `master`, when https://github.com/CartoDB/deep-insights.js/pull/562 is merged

---

cartodb PR: https://github.com/CartoDB/cartodb/pull/12543
deep-insights.js PR: https://github.com/CartoDB/deep-insights.js/pull/566
cartodb.js PR: https://github.com/CartoDB/cartodb.js/pull/1709

---

pulling the changes from time-series aggregations, and timezones, plus:

- [x] implemented responsive behaviour for Tablet and Mobile viewports
- [x] added timeslider tip to timeslider view, and static tips to time-slider header
- [x] added viewport utils
- [x] tests

there's only one thing I'm not specially happy about which is updating the static tips, when dragging the handles, however I couldn't find a better approach using the rangeFilter, as it is only updated on brush end

**acceptance**

- test time-series, animated time-series, AND normal histogram in desktop, tablet, and mobile viewport, selecting range, playing, and updating the timezone and local timezone for timestamps
- heights should vary
- timestamps should vary, and update on dragging and click interactions
- tips should change behaviour

test again with `number` type column, they should work as expected

check the Invision project for the designs

### CODE REVIEW

**Comment by matallo** [comment]
advanced a bit on the testing front, I feel the lack of specs disturbing

- [ ] _updateTriangle
- [ ] _updateAxisTip
- [ ] _onChangeDragging
- [ ] _toggleAxisTip
- [ ] _hideAxisTip
- [ ] _showAxisTip
- [x] _generateChartContent
- [ ] _setupBrush
- [x] _generateAxisTip
- [x] _isTabletViewport
- [ ] _generateHandle
- [ ] _updateChart
- [x] _isMobileViewport
- [ ] _generateBars

**Comment by matallo** [comment]
finished the specs, I didn't test the X position for the tip at the beginning / middle / end of the timeline, just the Y because it could be crazytown (and it wasn't actually the scope of the PR, just for the timeslider tip)

File: spec/widgets/histogram/chart.spec.js [hunk 1]
```diff
@@ -3,10 +3,24 @@ var $ = require('jquery');
...
 var d3 = require('d3');
 var WidgetHistogramChart = require('../../../src/widgets/histogram/chart');
+var viewportUtils = require('../../../src/viewport-utils');
+var formatter = require('../../../src/formatter');
+var moment = require('moment');
+require('moment-timezone');
+
+function flushAllD3Transitions () {
+  var now = Date.now;
+  Date.now = function () { return Infinity; };
```
Comment (matallo) [inline spec/widgets/histogram/chart.spec.js]: 😂

**Comment by matallo** [comment]
retest this, please

File: src/widgets/histogram/chart.js [hunk 2]
```diff
@@ -1023,8 +1045,12 @@ module.exports = cdb.core.View.extend({
   _generateAxisTip: function (className) {
     var handle = this.chart.select('.CDB-Chart-handle.CDB-Chart-handle-' + className);
 
-    var yPos = className === 'right' ? this.chartHeight() + (TRIANGLE_HEIGHT * TRIANGLE_RIGHT_FACTOR) : -(TRIANGLE_HEIGHT + TIP_RECT_HEIGHT + TOOLTIP_MARGIN);
-    var yTriangle = className === 'right' ? this.chartHeight() + (TRIANGLE_HEIGHT * TRIANGLE_RIGHT_FACTOR) + 2 : -(TRIANGLE_HEIGHT + TOOLTIP_MARGIN) - 2;
+    var yPos = className === 'right' && !(this._isMobileViewport() && this._isTimeSeries())
```
Comment (nobuti) [inline src/widgets/histogram/chart.js]: This code block is repeated above as well, it might be useful to extract it to its own method.

**Comment by nobuti** [comment]
Was this accepted?

**Comment by rubenmoya** [comment]
Yes, the last changes was just one test failing in Travis, but code review and acceptance were already done.

**Pull request merged**